recipe-java-runtime/src/main/java/ca/derekcormier/recipe/Oven.java

```diff
@@ -30,10 +31,13 @@ private Cake _bake(Recipe recipe, Cake cake) {
             List<Recipe.Segment> segments = recipe.segment();
             for (Recipe.Segment segment: segments) {
                 String payload = serializePayload(segment.recipe, cake);
-                for (Dispatcher dispatcher: dispatchers) {
-                    String jsonCake = dispatcher.dispatch(segment.domain, payload);
-                    cake = deserializeCake(jsonCake);
+
+                if (!dispatchers.containsKey(segment.domain)) {
+                    throw new RuntimeException("cannot dispatch ingredient; no dispatcher registered for domain '" + segment.domain + "'");
                 }
+
+                String jsonCake = dispatchers.get(segment.domain).dispatch(payload);
+                cake = deserializeCake(jsonCake);
```

Comment:
I'm still confused why the `cake` variable gets overridden in a loop. We only care about the last value?

Reply:
Cake starts with a value, gets serialized and passed to the dispatcher. The dispatcher returns the original cake plus what gets added to it during backing, re-serializes and dispatches to the next service, and so forth. Basically, you need to constantly keep passing the cake around to each service so that baking can read from and write to it.

Reply:
oh... because we serialize it into the payload? So the `payload = the segment.recipe + cake-so-far`?

Reply:
Yup exactly.

Reply:
The service does the baking, so it may update the cake with new keys.